rtmp-client/src/main/java/net/butterflytv/rtmp_client/RtmpClient.java

```diff
@@ -16,44 +16,262 @@
     private final static int TIMEOUT_IN_MS = 10000;
     private long rtmpPointer = 0;
 
+    private int rtmpError = RtmpIOException.RTMP_ERROR_NONE;
+
     /** Socket send timeout value in milliseconds */
     private int sendTimeoutInMs = TIMEOUT_IN_MS;
     /** Socket receive timeout value in seconds */
     private int receiveTimeoutInMs = TIMEOUT_IN_MS;
 
     public static class RtmpIOException extends IOException {
 
+        /**
+         * No error
+         */
+        public final static int RTMP_ERROR_NONE = 0;
+
+        /**
+         * RTMP read has received an EOF or READ_COMPLETE from the server
+         */
+        public final static int RTMP_READ_DONE = -1;
+
         /**
          * it means there is a problem in memory allocation
          */
-        public final static int OPEN_ALLOC = -1;
+        public final static int OPEN_ALLOC = -2;
```

Comment:
Can we rephrase the Javadoc for this and the next 4 constants to be consistent with the ones you're adding, whilst we're editing this? I think the style you're using is nicer ("it means there" at the start of every comment doesn't add value :))

Reply:
Done. Also removed two unused error codes from previous version.